components/tabs/tab-mixin.js

```diff
@@ -0,0 +1,157 @@
+import '../colors/colors.js';
+import { css, html } from 'lit';
+import { SkeletonMixin } from '../skeleton/skeleton-mixin.js';
+
+const keyCodes = {
+	ENTER: 13,
+	SPACE: 32
+};
+
+export const TabMixin = superclass => class extends SkeletonMixin(superclass) {
+
+	static get properties() {
+		return {
+			selected: { type: Boolean, reflect: true },
+		};
+	}
+
+	static get styles() {
+		const styles = [ css`
+			:host {
+				box-sizing: border-box;
+				display: inline-block;
+				max-width: 200px;
+				outline: none;
+				position: relative;
+				vertical-align: middle;
+			}
+			.d2l-tab-selected-indicator {
+				border-top: 4px solid var(--d2l-color-celestine);
+				border-top-left-radius: 4px;
+				border-top-right-radius: 4px;
+				bottom: 0;
+				display: none;
+				margin: 1px 0.6rem 0 0.6rem;
+				position: absolute;
+				transition: box-shadow 0.2s;
+				width: calc(100% - 1.2rem);
+			}
+			:host(:first-child) .d2l-tab-selected-indicator {
+				margin-inline-end: 0.6rem;
+				margin-inline-start: 0;
+				width: calc(100% - 0.6rem);
+			}
+			:host([aria-selected="true"]:focus) {
+				text-decoration: none;
+			}
+			:host(:hover) {
+				color: var(--d2l-color-celestine);
+				cursor: pointer;
+			}
+			:host([aria-selected="true"]:hover) {
+				color: inherit;
+				cursor: default;
+			}
+			:host([aria-selected="true"]) .d2l-tab-selected-indicator {
+				display: block;
+			}
+		`];
+
+		super.styles && styles.unshift(super.styles);
+		return styles;
+	}
+
+	constructor() {
+		super();
+		this.ariaSelected = false;
+		this.role = 'tab';
+		this.selected = false;
+		this.tabIndex = -1;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		this.#addEventHandlers();
+
+		if (!this.#resizeObserver) {
+			this.#resizeObserver = new ResizeObserver(() => {
+				this.#handleResize();
+			});
+			this.#resizeObserver.observe(this);
+		}
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		this.#removeEventHandlers();
+
+		if (this.#resizeObserver) {
+			this.#resizeObserver.disconnect();
+			this.#resizeObserver = null;
+		}
+	}
+
+	render() {
+		return html`
+			<div class="d2l-skeletize">
```

Comment:
Currently, `d2l-tab-wip` handles skeletizing its content. If we keep that approach, we can remove this  `d2l-skeletize` from this `div`. If we don't want to give the subclass control over how the tab content is skeletized, we'd want to keep this class and remove the skeleton stuff from `d2l-tab-wip`.

I think letting the subclass skeletized individual parts of its tab content could look a bit busy, but that's perhaps a conversation we should have with Jeff.

Reply:
If we end up removing this class here, I think we should keep this `div` though, because it is critical to the overall layout with respect to the indicator.